core/unit_test/view/TestBasicView.hpp

```diff
@@ -0,0 +1,232 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+#include <type_traits>
+#include <experimental/__p0009_bits/dynamic_extent.hpp>
+#include <experimental/__p2642_bits/layout_padded_fwd.hpp>
+#define KOKKOS_IMPL_PUBLIC_INCLUDE
+#include <View/MDSpan/Kokkos_MDSpan_Accessor.hpp>
+#include <View/Kokkos_BasicView.hpp>
+#undef KOKKOS_IMPL_PUBLIC_INCLUDE
+
+namespace {
+template <class IndexType, std::size_t>
+IndexType fill_zero() {
+  return IndexType(0);
+}
+
+template <class ExecutionSpace, class Extents>
+auto make_spanning_mdrange_policy_from_extents_impl(const Extents &extents,
+                                                    std::index_sequence<0>) {
+  return Kokkos::RangePolicy<ExecutionSpace>{0, extents.extent(0)};
+}
+
+template <class ExecutionSpace, class Extents, std::size_t... Indices>
+auto make_spanning_mdrange_policy_from_extents_impl(
+    const Extents &extents, std::index_sequence<Indices...>) {
+  using index_type    = typename Extents::index_type;
+  constexpr auto rank = Extents::rank();
+  return Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<rank>>{
+      {fill_zero<index_type, Indices>()...}, {extents.extent(Indices)...}};
+}
+
+template <class ExecutionSpace, class Extents>
+auto make_spanning_mdrange_policy_from_extents(const Extents &extents) {
+  return make_spanning_mdrange_policy_from_extents_impl<ExecutionSpace>(
+      extents, std::make_index_sequence<Extents::rank()>{});
+}
+
+template <class T, class ExecutionSpace>
+struct TestBasicView {
+  template <class ExtentsType>
+  static void test_default_constructor() {
+    using extents_type  = ExtentsType;
+    using layout_type   = Kokkos::Experimental::layout_right_padded<>;
+    using accessor_type = Kokkos::Impl::checked_reference_counted_accessor<
+        T, typename ExecutionSpace::memory_space>;
+    using view_type =
+        Kokkos::BasicView<T, extents_type, layout_type, accessor_type>;
+    view_type view;
+
+    EXPECT_FALSE(view.data_handle().has_record());
+    EXPECT_EQ(view.data_handle().get(), nullptr);
+    EXPECT_EQ(view.extents(), extents_type{});
+    EXPECT_EQ(view.data_handle().use_count(), 0);
+    EXPECT_TRUE(view.is_exhaustive());
+    EXPECT_EQ(view.data_handle().get_label(), "");
+    EXPECT_TRUE(view.empty());
+    EXPECT_EQ(view.size(), 0u);
+  }
```

Comment:
This pattern is outdated.
Please prefer a more fine grained approach
```C++

TEST(TEST_CATEGOTY, basic_view_default_constructor)
{
  test_default_constructor<...>();  // <- non static non member function template
}

Reply:
done